Sources/GRPCHTTP2TransportNIOPosix/GRPCHTTP2TransportNIOPosix.swift

```diff
@@ -15,7 +15,88 @@
  */
 
 import GRPCCore
+import GRPCHTTP2Core
+import NIOCore
+import NIOExtras
+import NIOPosix
 
-@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
-public struct GRPCHTTP2TransportNIOPosix {
+@available(macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0, *)
+public struct GRPCHTTP2TransportNIOPosix: ServerTransport {
```

Comment:
We don't need a GRPC prefix here. 

To help give some structure I've used some namespacing for the client, see https://github.com/grpc/grpc-swift/pull/1880/files.

For the server we should do similar:

```swift
public enum HTTPServerTransport {}

extension HTTPServerTransport {
  public struct Posix: ServerTransport { ... }
}

extension HTTPServerTransport.Posix {
  public struct Config {
    // ...
  }
}
```

I think it makes sense gathering them like this as they are _so_ closely related.